src/lib/evil/evil_libgen.h

```diff
@@ -3,12 +3,19 @@
 
 #include <evil_private.h>
 
+#ifndef HAVE_BASENAME
+# define HAVE_BASENAME
+# define basename evil_basename
+#endif
+
 #ifndef HAVE_DIRNAME
 # define HAVE_DIRNAME
 # define dirname evil_dirname
 #endif
 
-EVIL_API char* dirname (char* path);
+EVIL_API char* evil_basename (char* path);
+
+EVIL_API char* evil_dirname (char* path);
```

Comment:
Hm, you could've just alternate them, but ok haha

Reply:
yeah but it is fixing this too: `dirname` -> `evil_dirname`

Reply:
I think he says something like:

```
#ifndef HAVE_BASENAME
# define HAVE_BASENAME
# define basename evil_basename
#endif

EVIL_API char* evil_basename (char* path);

#ifndef HAVE_DIRNAME
# define HAVE_DIRNAME
# define dirname evil_dirname
#endif

EVIL_API char* evil_dirname (char* path);
```